python/mcap/mcap/mcap0/reader.py

```diff
@@ -251,7 +266,29 @@ def iter_messages(
             timestamp are not included.
         :param end_time: an integer nanosecond timestamp. if provided, messages logged after this
             timestamp are not included.
+        :param log_time_order: if True, messages will be yielded in order of their log times. If
+            False, messages will be yielded in the order they appear in the MCAP file.
+
+        .. warning::
+            setting log_time_order to True on a non-seekable stream will cause the entire content
+            of the MCAP to be loaded into memory.
         """
+        if not log_time_order:
+            for t in self._iter_messages_internal(topics, start_time, end_time):
+                yield t
+        else:
+            for t in sorted(
+                self._iter_messages_internal(topics, start_time, end_time),
+                key=lambda tup: tup[2].log_time,
```

Comment:
```suggestion
                key=lambda (schema, channel, message): message.log_time,
```

Reply:
Pyright currently struggles to handle this syntax, not sure what's happening.

Reply:
```
mcap/tests/test_reader.py:7: in <module>
    from mcap.mcap0.reader import make_reader, SeekingReader, NonSeekingReader, McapReader
E     File "/Users/j/code/mcap/python/mcap/mcap/mcap0/reader.py", line 291
E       key=lambda (schema, channel, message): message.log_time,
E                  ^
E   SyntaxError: invalid syntax
```